src/Microsoft.OData.Core/UriParser/Parsers/FunctionCallParser.cs

```diff
@@ -60,11 +77,15 @@ public ExpressionLexer Lexer
         /// Parses an identifier that represents a function.
         /// </summary>
         /// <param name="parent">Token for the parent of the function being parsed.</param>
-        /// <returns>QueryToken representing this function.</returns>
-        public QueryToken ParseIdentifierAsFunction(QueryToken parent)
+        /// <param name="result">QueryToken representing this function.</param>
+        /// <returns>True if the parsing was successful.</returns>
+        public bool ParseIdentifierAsFunction(QueryToken parent, out QueryToken result)
         {
+            result = null;
             string functionName;
 
+            lexer.TakeTextPositionSnapshot();

```

Comment:
It's possible, but it will make the code a lot more messier.
The problem is that we don't have a way of telling the difference between a function call and a aggregate of a entity set property without looking at the arguments.
This happens because, although function calls must contain a namespace separated by a dot of the function identifier by OData V4 spec, our stack has an extension that allows to call a function without the namespace.

Thus the syntax is only different if you look at the arguments. To parse the aggregation clause we use a `UriQueryExpressionParser` and the `Lexer`, both keep state of the string being parsed. When parsing the expressions of the aggregation clause, the `UriQueryExpressionParser` creates a `IdentifierTokenizer` that takes a `FunctionCallParser` as a parameter. 

The `FunctionCallParser` is the "ultimate responsible" to tell if the text corresponds to a function call. It can only tell if looking at the parameters. To do it have to advance the parser. If it finds out that is indeed a entity set aggregation the ideal thing to do would be to parse it. But the class that have this logic is the `UriQueryExpressionParser`, what makes it troublesome to reuse the code and make the logic in only one place.

That's why I decided to snapshot the parser. I was reluctant of doing it, but was the best approach I found to not duplicate code or create a spaghetti.